### PR TITLE
[WIP] Improved node compilation in CompileNodeToValue (eval solution)

### PR DESCRIPTION
**Pull request by asgrim**
This PR aims to solve both #64 and #19 

An alternative solution exists in #95

### CODE REVIEW

**Comment by asgrim** [comment]
@Ocramius thoughts? :P

File: src/NodeCompiler/CompileNodeToValue.php [hunk 1]
```diff
@@ -15,51 +16,67 @@ class CompileNodeToValue
...
-                    return true;
-                default:
-                    // @todo this should evaluate the VALUE, not the name
-                    /* @see https://github.com/Roave/BetterReflection/issues/19 */
-                    return $firstName;
-            }
+        $printer = new Printer();
+        $code = $printer->prettyPrint([$node]);
+
+        eval('$x = ' . $code);
```
Comment (nikic) [inline src/NodeCompiler/CompileNodeToValue.php]: A constant access could be nested in an expression (`foo::bar + 1`), in which case eval won't work (unless you actually want a runtime constant lookup).

Comment (Ocramius) [inline src/NodeCompiler/CompileNodeToValue.php]: Traversal to replace such nodes should indeed happen upfront :-(

Comment (asgrim) [inline src/NodeCompiler/CompileNodeToValue.php]: Aye, ConstFetch and ClassConstFetch have dedicated handlers above (although I haven't dealt with them properly yet still). It's mainly my concern using `eval` tbh...

Comment (Ocramius) [inline src/NodeCompiler/CompileNodeToValue.php]: @asgrim you can probably use `eval()` if you first validate that ALL the nodes are within a given whitelist (WHITELIST! No blacklists, plox :P )

Comment (asgrim) [inline src/NodeCompiler/CompileNodeToValue.php]: thanks for feedbacks folks :)

**Pull request closed without merging**